### DISCUSSION TITLE
Advice on model selection

### DISCUSSION THREAD

**Discussion by simonziegler** [betalgo/openai#364] [Q&A]
I am revisiting some code I wrote in Q1 this year, where I use `Models.TextDavinciV3` to create some code from a prompt. I am looking to update beyond GPT-3, and when I use either `Models.Gpt_3_5_Turbo` or `Models.Gpt_4` I get an error as below. What model should I use please?

```
System.Exception: This is a chat model and not supported in the v1/completions endpoint. Did you mean to use v1/chat/completions?
```

**Comment by kayhantolga** [comment]
`Models.TextDavinciV3` is for `completion API`
`Models.Gpt_3_5_Turbo` and `Models.Gpt_4` is for `chat API`.
More information please visit this documentation:
https://platform.openai.com/docs/models/model-endpoint-compatibility
